charts/argo-rollouts/templates/controller/role.yaml

```diff
@@ -8,290 +8,5 @@ metadata:
-- apiGroups:
-  - batch
-  resources:
-  - jobs
-  verbs:
-  - create
-  - get
-  - list
-  - watch
-  - update
-  - patch
-  - delete
-{{- if .Values.providerRBAC.enabled }}
-{{- if .Values.providerRBAC.providers.istio }}
-# virtualservice/destinationrule access needed for using the Istio provider
-- apiGroups:
-  - networking.istio.io
-  resources:
-  - virtualservices
-  - destinationrules
-  verbs:
-  - watch
-  - get
-  - update
-  - patch
-  - list
-{{- end }}
-{{- if .Values.providerRBAC.providers.smi }}
-# trafficsplit access needed for using the SMI provider
-- apiGroups:
-  - split.smi-spec.io
-  resources:
-  - trafficsplits
-  verbs:
-  - create
-  - watch
-  - get
-  - update
-  - patch
-{{- end }}
-{{- if .Values.providerRBAC.providers.ambassador }}
-# ambassador access needed for Ambassador provider
-- apiGroups:
-  - getambassador.io
-  - x.getambassador.io
-  resources:
-  - mappings
-  - ambassadormappings
-  verbs:
-  - create
-  - watch
-  - get
-  - update
-  - list
-  - delete
-{{- end }}
-{{- if .Values.providerRBAC.providers.awsLoadBalancerController }}
-# Endpoints and TargetGroupBindings needed for ALB target group verification when using AWS Load Balancer Controller
-- apiGroups:
-  - ""
-  resources:
-  - endpoints
-  verbs:
-  - get
-- apiGroups:
-  - elbv2.k8s.aws
-  resources:
-  - targetgroupbindings
-  verbs:
-  - list
-  - get
-{{- end }}
-{{- if .Values.providerRBAC.providers.awsAppMesh }}
-# AppMesh virtualservices/virtualrouter CRD read-only access needed for using the App Mesh provider
-- apiGroups:
-  - appmesh.k8s.aws
-  resources:
-  - virtualservices
-  verbs:
-  - watch
-  - get
-  - list
-# AppMesh virtualnode CRD r/w access needed for using the App Mesh provider
-- apiGroups:
-  - appmesh.k8s.aws
-  resources:
-  - virtualnodes
-  - virtualrouters
-  verbs:
-  - watch
-  - get
-  - list
-  - update
-  - patch
-{{- end }}
-{{- if .Values.providerRBAC.providers.traefik }}
-# Traefik access needed when using the Traefik provider
-- apiGroups:
-  - traefik.containo.us
-  - traefik.io
-  resources:
-  - traefikservices
-  verbs:
-  - watch
-  - get
-  - update
-{{- end }}
-{{- if .Values.providerRBAC.providers.apisix }}
-# Access needed when using the Apisix provider
-- apiGroups:
-  - apisix.apache.org
-  resources:
-  - apisixroutes
-  verbs:
-  - watch
-  - get
-  - update
-{{- end }}
-{{- if .Values.providerRBAC.providers.glooPlatform }}
-  # Access needed when using the Gloo Platform provider
-- apiGroups:
-  - networking.gloo.solo.io
-  resources:
-  - routetables
-  verbs:
-  - '*'
-{{- end }}
-{{- if .Values.providerRBAC.providers.gatewayAPI }}
-  # Access needed when using the Gateway API provider
-- apiGroups:
-  - gateway.networking.k8s.io
-  resources:
-  - httproutes
-  - tcproutes
-  - tlsroutes
-  - udproutes
-  - grpcroutes
-  verbs:
-  - get
-  - list
-  - watch
-  - update
-{{- end }}
-{{- end }}
+  {{- include "argo-rollouts.controller.roleRuls" . | nindent 2 }}
```

Comment:
Clusterrole has `{{- with .Values.providerRBAC.additionalRules }}` block and Role doesn't have it, so I thought it's better to merge into a single definition.

<img width="518" alt="image" src="https://github.com/user-attachments/assets/f1d9338b-2176-4546-8c66-23a1f30aff12">

Reply:
Thank you for reviewing. Fixed in https://github.com/argoproj/argo-helm/pull/2830/commits/eba2e52bfd25a15ae5e97144e27a367179b2b16f 😇